### ISSUE TITLE
make nn.CrossEntropyCriterion stateless

### ISSUE THREAD

**Issue by iamalbert** [torch/nn#807]
make nn.CrossEntropyCriterion stateless to support direct `backward` call

**Comment by apaszke** [comment]
Well, it's a reasonable requirement. It's clearly stated in the documentation that you should call backward, after calling forward, with the same input. Otherwise it's perfectly valid for the module to compute bad gradients.

**Comment by yangky11** [comment]
And it's the case for not only CrossEntropyCritetion but also many modules such as SpatialConvolution
